File: src/main/java/br/com/tech/challenge/service/validation/vote/SessionAlreadyClosed.java
package br.com.tech.challenge.service.validation.vote;


import br.com.tech.challenge.domain.VoteDetail;
import br.com.tech.challenge.exceptions.SessionAlreadyClosedException;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Service;

@Service
@Order(1)
public class SessionAlreadyClosed implements VoteValidation {
    @Override
    public void validate(VoteDetail voteDetail) {
        if(!voteDetail.session().isOpenToVote()){
            throw new SessionAlreadyClosedException(STR."Session with {id} = \{voteDetail.session().getId()} closed at \{voteDetail.session().getFinishAt()}");
        }
    }
}

File: src/main/java/br/com/tech/challenge/service/validation/vote/SessionAlreadyClosed.kt
package br.com.ntconsult.challenge.service.validation.vote

import br.com.ntconsult.challenge.domain.VoteDetail
import br.com.ntconsult.challenge.exception.SessionAlreadyClosedException
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Service

@Service
@Order(1)
class SessionAlreadyClosed: VoteValidation {
    override fun valid(voteDetail: VoteDetail) {
        if(!voteDetail.session.isOpenToVote()){
            throw SessionAlreadyClosedException("Session with {id} = ${voteDetail.session.id} closed at ${voteDetail.session.finishAt}")
        }
    }
}